lib/rubocop/cop/bundler/gem_version_declaration.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Bundler
+      # Enforce that Gem version declarations are either required
+      # or prohibited.
+      #
+      # @example EnforcedStyle: required (default)
+      #  # bad
+      #  gem 'rubocop'
+      #
+      #  # good
+      #  gem 'rubocop', '~> 1.12'
+      #
+      #  # good
+      #  gem 'rubocop', '>= 1.10.0'
+      #
+      #  # good
+      #  gem 'rubocop', '>= 1.5.0', '< 1.10.0'
+      #
+      # @example EnforcedStyle: prohibited
+      #  # good
+      #  gem 'rubocop'
+      #
+      #  # bad
+      #  gem 'rubocop', '~> 1.12'
+      #
+      #  # bad
+      #  gem 'rubocop', '>= 1.10.0'
+      #
+      #  # bad
+      #  gem 'rubocop', '>= 1.5.0', '< 1.10.0'
+      #
+      class GemVersionDeclaration < Base
+        include ConfigurableEnforcedStyle
+
+        REQUIRED_MSG = 'Gem version declaration is required.'
+        PROHIBITED_MSG = 'Gem version declaration is prohibited.'
+        VERSION_DECLARATION_REGEX = /^[~<>=]*\s?[0-9.]+/.freeze
```

Comment:
```suggestion
        VERSION_DECLARATION_REGEX = /^\s*[~<>=]*\s*[0-9.]+/.freeze
```

Multiple spaces are valid in bundler before and after the operator. We might also want to be more specific about the format here, but it probably doesn't _really_ matter.

Reply:
I've added the suggested change.

> We also could be more specific about the format here

Do you mind detailing what you mean by this? Or is it the change you provided?

Reply:
What I mean is the Regexp will match `......` or `>>>>> 7` or other random strings that aren’t version specifications, but again it probably doesn't really matter. 

It could be more strict with something like `^\s*(?:~>|[<>]?=?)?\s*[0-9]+(\.[0-9]+)*`